code/game/objects/items/devices/radio/electropack.dm

```diff
@@ -222,3 +223,82 @@ Code:
 			onclose(user, "radio")
 			return
 
+/obj/item/device/electropack/shockcollar/proc/remove_id()
+	if (id && ismob(loc))
+		var/mob/M = loc
+		M.put_in_hands(id)
+		to_chat(usr, "<span class='notice'>You remove the ID from the [name].</span>")
+	else
+		id.forceMove(drop_location())
+	id = null
+
+/obj/item/device/electropack/shockcollar/verb/verb_remove_id()
+	set category = "Object"
+	set name = "Remove ID"
+	set src in usr
+
+	if(issilicon(usr))
+		return
+
+	if (usr.canUseTopic(src))
+		if(id)
+			remove_id()
+		else
+			to_chat(usr, "<span class='warning'>This collar does not have an ID in it!</span>")
+
+/obj/item/device/electropack/shockcollar/proc/id_check(mob/user, obj/item/card/id/I)
+	if(!I)
+		if(id)
+			remove_id()
+			return TRUE
+		else
+			var/obj/item/card/id/C = user.get_active_held_item()
+			if(istype(C))
+				I = C
+
+	if(I)
+		if(!user.transferItemToLoc(I, src))
+			return FALSE
+		var/obj/old_id = id
```

Comment:
replace this entire block with
`
if(old_id)
    if(!user.put_in_hands(old_id))
        old_id.forceMove(drop_location())
    id = I
`

Reply:
formatting's fucked up by github but you get the point